context/app/static/js/components/cell-types/CellTypesBiomarkersTable.tsx

```diff
@@ -1,25 +1,43 @@
 import React from 'react';
-import SectionHeader from 'js/shared-styles/sections/SectionHeader';
-import Description from 'js/shared-styles/sections/Description';
 import Select from '@mui/material/Select';
 import MenuItem from '@mui/material/MenuItem';
 import Stack from '@mui/material/Stack';
-import { useTabs } from 'js/shared-styles/tabs';
-import { Tab, Tabs, TabPanel } from 'js/shared-styles/tables/TableTabs';
 import Table from '@mui/material/Table';
 import TableHead from '@mui/material/TableHead';
-import { FormControl, InputLabel, Paper, TableBody, TableCell, TableRow } from '@mui/material';
+import FormControl from '@mui/material/FormControl';
+import InputLabel from '@mui/material/InputLabel';
+import Paper from '@mui/material/Paper';
+import TableBody from '@mui/material/TableBody';
+import TableCell from '@mui/material/TableCell';
+import TableRow from '@mui/material/TableRow';
+import { capitalize } from '@mui/material/utils';
+
+import SectionHeader from 'js/shared-styles/sections/SectionHeader';
+import Description from 'js/shared-styles/sections/Description';
+import { useTabs } from 'js/shared-styles/tabs';
+import { Tab, Tabs, TabPanel } from 'js/shared-styles/tables/TableTabs';
 import { StyledTableContainer } from 'js/shared-styles/tables';
 import { InternalLink } from 'js/shared-styles/Links';
 import { CellTypeBiomarkerInfo } from 'js/hooks/useUBKG';
-import { useCellTypeBiomarkers } from './hooks';
+
 import { DetailPageSection } from '../detailPage/style';
+import { useCellTypeBiomarkers } from './hooks';
 
+const tableKeys = ['genes', 'proteins'] as const;
+type TableKey = (typeof tableKeys)[number];
 interface BiomarkerTableProps {
-  tableKey: 'genes' | 'proteins';
+  tableKey: TableKey;
 }
 
-function BiomarkerTableRow({ biomarker, type }: { biomarker: CellTypeBiomarkerInfo; type: 'genes' | 'proteins' }) {
+function otherTableKey(tableKey: TableKey): TableKey {
+  const otherKey = tableKeys.find((key) => key !== tableKey);
+  if (!otherKey) {
+    throw new Error(`Could not find other table key for ${tableKey}. This should never happen.`);
+  }
+  return otherKey;
+}
```

Comment:
This will get picked up by the route level error boundary? Should we handle it more gracefully in case it ever does happen?

Reply:
This is a type guard function used to safely access `biomarkers.genes` or `biomarkers.proteins` below, it should never reach the error case since those keys should always be present. This was added in an effort to deduplicate the two tables' code per your previous suggestion.

These tables are also not currently functional due to the cells API not having this functionality hooked up and the UBKG API no longer caching this data for us.